includes/Handlers/Tools/ToolsHandler.php

```diff
@@ -133,6 +133,15 @@ public function call_tool( array $message, int $request_id = 0 ): array {
 				),
 			);
 
+			// Extract and store metadata before adding result to response content.
+			$response['_metadata'] = $result['_metadata'] ?? array(
+				'component_type' => 'tool',
+				'tool_name'      => $request_params['name'],
+			);
+
+			// Remove metadata from result so it doesn't appear in content or structuredContent.
+			unset( $result['_metadata'] );
```

Comment:
I'm really confused by this. We're storing the metadata in the response, removing it from the result, and then returning the response (that now has the metadata)? Doesn't that mean we _are_ putting the metadata in the response? 😕

Reply:
Oh my. Well, I'm glad it works for now! Looking forward to some DTOs! 😅 👍